packages/framework/tests/Unit/Component/Money/MoneyTest.php

```diff
@@ -0,0 +1,582 @@
+        yield ['0.0', '0.0'];
+        yield ['0.00', '0.00'];
+        yield ['000', '0'];
+        yield ['000.0', '0.0'];
+        yield ['010', '10'];
+        yield ['1e1', '10'];
+    }
+
+    /**
+     * @dataProvider invalidValuesFromStringProvider
+     * @param string $string
+     */
+    public function testInvalidValuesInFromString(string $string): void
+    {
+        $this->expectException(BigNumbersError::class);
+
+        Money::fromString($string);
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function invalidValuesFromStringProvider(): Iterator
+    {
+        yield [''];
+        yield ['xxx'];
+        yield ['1,00'];
+        yield ['0.'];
+        yield ['.0'];
+        yield ['1.0.0'];
+        yield [' 0'];
+        yield ['0 '];
+        yield ['1+1'];
+        yield ['1 000'];
+        yield ['0x0'];
+        yield ['+-1'];
+        yield ['++1'];
+        yield ['--1'];
+    }
+
+    /**
+     * @dataProvider fromIntegerProvider
+     * @param int $integer
+     * @param string $expected
+     */
+    public function testFromInteger(int $integer, string $expected): void
+    {
+        $money = Money::fromInteger($integer);
+
+        $this->assertEquals($expected, $money->toString());
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function fromIntegerProvider(): Iterator
+    {
+        yield [0, '0'];
+        yield [1, '1'];
+        yield [-1, '-1'];
+        yield [10, '10'];
+        yield [PHP_INT_MAX, (string)PHP_INT_MAX];
+    }
+
+    /**
+     * @dataProvider invalidValuesFromFloatProvider
+     * @param float $float
+     * @param int $scale
+     */
+    public function testInvalidValuesInFromFloat(float $float, int $scale): void
+    {
+        $this->expectException(BigNumbersError::class);
+
+        Money::fromFloat($float, $scale);
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function invalidValuesFromFloatProvider(): Iterator
+    {
+        yield [NAN, 0];
+        yield [INF, 0];
+        yield [-INF, 0];
+        yield [NAN, 1];
+        yield [INF, 1];
+        yield [-INF, 1];
+    }
+
+    /**
+     * @dataProvider fromFloatProvider
+     * @param float $float
+     * @param int $scale
+     * @param string $expected
+     */
+    public function testFromFloat(float $float, int $scale, string $expected): void
+    {
+        $money = Money::fromFloat($float, $scale);
+
+        $this->assertEquals($expected, $money->toString());
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function fromFloatProvider(): Iterator
+    {
+        yield [0, 0, '0'];
+        yield [0, 1, '0.0'];
+        yield [0, 10, '0.0000000000'];
+        yield [1, 0, '1'];
+        yield [-1, 0, '-1'];
+        yield [10, 0, '10'];
+        yield [0.05, 1, '0.1'];
+        yield [0.5, 0, '1'];
+        yield [0.0001, 3, '0.000'];
+        yield [1.0001, 3, '1.000'];
+    }
+
+    public function testAddIsImmutable(): void
```

Comment:
What about mergin all immutable tests into one `testImmutability`?
1. Create Money Object
2. Perform all operations
3. Test not changed

Reply:
We do not see any advantage of such solution, but we see one disadvantage and that is that you would not see which test method failed.